Editors/CodeEditor.h

```diff
@@ -0,0 +1,22 @@
+#ifndef CODEEDITOR_H
+#define CODEEDITOR_H
+
+#include "BaseEditor.h"
+#include "ui_CodeEditor.h"
+
+namespace Ui {
+class CodeEditor;

```

Comment:
This is partly a mistake I made and just fixed: 7080724b46950a4e97245ae5f917afed8f33b2b8

Because ScriptEditor inherits CodeEditor, it needs the complete type of Ui::CodeEditor available in its constructor for widget data mapping. Hence it needs to include `ui_CodeEditor.h` somewhere to get the complete type. My previous work around was to just include `ui_CodeEditor.h` from `CodeEditor.h` but that's a bad idea because it pollutes more.

Everything in regards to the class naming conventions is just the Qt default. We can discuss changing the projects naming conventions later, but I didn't really want to deviate from typical standard Qt conventions (because that could really bite us).